### DISCUSSION TITLE
Can't Find My Express API When Deploying To Vercel

### DISCUSSION THREAD

**Discussion by tubbo** [vercel/community#644] [Help]
I'm trying to deploy an existing ExpressJS API to Vercel within a monorepo. There are two top-level directories, `api/` and `web/`, that I'm trying to deploy to two Vercel projects. I have the projects set up and they both say they deploy green to production. So far, so good.

For the API, I read that I need to have my functions in an `api/` subdirectory of my root dir, which is `api/`, so I assume that means my functions need to be in `$REPO_ROOT/api/api`? Since my Express app is built with TypeScript, I changed the `outDir` to point to `./api` instead of dist, that way Vercel _should_ be able to find my entrypoint at least.

But when I visit https://my-api.vercel.app/api, I get a 404. I've tried https://my-api.vercel.app/api/api and https://my-api.vercel.app/api/api/some-known-route but neither of them work. I get 404s every time.

The only thing I _don't_ get a 404 for is a robots.txt file I dropped in public just so Vercel will not yell at me for having a non-empty public dir.

I'm not sure where to proceed from here. Vercel doesn't show me my actual build dir so I can't see if everything worked out. Locally when I build with that output dir I see an `api/server.js` so logically if I go to https://my-api.vercel.app/api/server shouldn't that serve my API? It seems like it's at least possible to run an ExpressJS app on Vercel, as I see other examples (albeit some of them using deprecated config that doesn't seem to work for me), but I don't really know what I'm doing wrong.

**Comment by leerob** [comment]
Hey! Have you seen this by chance? 

https://vercel.com/guides/using-express-with-vercel

**Reply by tubbo** [reply]
Yes, I read that guide and I have that `vercel.json` in `$REPO_ROOT/api/vercel.json`.

**Reply by tubbo** [reply]
Should be noted that in my build logs I get this at the end:

```
Generated build outputs:
--
16:29:06.778 | - Static files: 1
16:29:06.778 | - Serverless Functions: 0
16:29:06.778 | - Edge Functions: 0
```

I don't think Vercel is seeing my API code...Does Vercel do any kind of analysis to make sure the JS files in `api/` are actually "Serverless Functions" or does it just count every JS file in that dir as a function?

I am building my TypeScript app in the build phase, these files aren't just there in the repo by default...could that be part of the problem?

This is my build command, by the way:

```bash
cd ../ && yarn generate && yarn workspace api build --outDir api
```
